src/app/shared/courses/courses.service.ts

```diff
@@ -14,7 +16,11 @@ const httpOptions = {
 export class CoursesService {
     courses: Course[] = mockedCourses;
 
-    constructor(public orderByName: OrderByPipe, private http: HttpClient){}
+    constructor(
+        public orderByName: OrderByPipe,
+        private http: HttpClient,
+        private preloadingService: PreloadingService
```

Comment:
I'd suggest naming it `LoaderService` as it actually does preload anything from API.

Reply:
it has become redundant. deleted